api/v1beta1/azuremachinetemplate_webhook.go

```diff
@@ -35,41 +39,54 @@ const (
 // SetupWebhookWithManager sets up and registers the webhook with the manager.
 func (r *AzureMachineTemplate) SetupWebhookWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewWebhookManagedBy(mgr).
-		For(r).
+		For(&AzureMachineTemplate{}).
```

Comment:
```suggestion
		For(r).
```
nit: Doesn't really matter but you can re-use `r` here

Reply:
sgtm!